### PR TITLE
FB-116 add formatting methods

### LINKED ISSUE #2529: Add writef, writefln methods to Outputs
Proposition. Asked for by some.

```php
writef($format, $arguments...)
writefln($format, $arguments...)
```

### PR DESCRIPTION
**Pull request by fvovan**
closes #2529

### CODE REVIEW

**Comment by fvovan** [comment]
should we leave them inside `CM_OutputStream_Abstract` (rather than including methods to `CM_OutputStream_Interface` )? 

@tomaszdurka wdyt?

**Comment by fvovan** [comment]
okay, Mocka does not support variadic arguments yet. will create PR there first :)

File: library/CM/OutputStream/Abstract.php [hunk 1]
```diff
@@ -8,4 +8,20 @@
...
+    public function writef($format, ...$params) {
+        $this->write(sprintf($format, ...$params));
+    }
+
+    /**
+     * @param string $format
+     * @param array  ...$params
+     */
+    public function writefln($format, ...$params) {
+        $this->writef(sprintf($format . PHP_EOL, ...$params));
```
Comment (tomaszdurka) [inline library/CM/OutputStream/Abstract.php]: I think you don't need that sprintf here?!

Either:
```php
$this->writef($format . PHP_EOL, ...$params);
// or
$this->writeln(sprintf($format . PHP_EOL, ...$params));
```
right?

Comment (fvovan) [inline library/CM/OutputStream/Abstract.php]: ah! good catch!

**Comment by fvovan** [comment]
waiting for https://github.com/tomaszdurka/codegenerator/pull/21

**Pull request merged**